src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1475 @@
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+# include <Inventor/actions/SoGetBoundingBoxAction.h>
+# include <Inventor/nodes/SoCamera.h>
+# include <Inventor/nodes/SoClipPlane.h>
+# include <Inventor/nodes/SoGroup.h>
+# include <Inventor/nodes/SoOrthographicCamera.h>
+# include <Inventor/sensors/SoTimerSensor.h>
+# include <QDialog>
+# include <QDockWidget>
+# include <QPointer>
+//# include <cmath>
+#endif
+
+#include "SectionCutting.h"
+#include "ui_SectionCutting.h"
+#include <App/Link.h>
+#include <App/Part.h>
+#include <Base/UnitsApi.h>
+#include <Gui/Application.h>
+#include <Gui/Command.h>
+#include <Gui/DockWindowManager.h>
+#include <Gui/Document.h>
+#include <Gui/MainWindow.h>
+#include <Gui/View3DInventor.h>
+#include <Gui/View3DInventorViewer.h>
+#include <Mod/Part/App/FeatureCompound.h>
+#include <Mod/Part/App/FeaturePartBox.h>
+#include <Mod/Part/App/FeaturePartCommon.h>
+#include <Mod/Part/App/FeaturePartCut.h>
+#include <Mod/Part/App/FeaturePartFuse.h>
+#include <Mod/Part/App/PartFeatures.h>
+
+using namespace PartGui;
+
+enum Refresh : bool
+{
+    notXValue = false,
+    notYValue = false,
+    notZValue = false,
+    notXRange = false,
+    notYRange = false,
+    notZRange = false,
+    XValue = true,
+    YValue = true,
+    ZValue = true,
+    XRange = true,
+    YRange = true,
+    ZRange = true
+};
+
+SectionCut::SectionCut(Gui::View3DInventor* view, QWidget* parent)
+    : QDialog(parent)
+    , ui(new Ui_SectionCut)
+{
+    // create widgets
+    ui->setupUi(this);
+    ui->cutX->setRange(-INT_MAX, INT_MAX);
+    ui->cutY->setRange(-INT_MAX, INT_MAX);
+    ui->cutZ->setRange(-INT_MAX, INT_MAX);
+
+    view = view;
+    Gui::View3DInventorViewer* viewer = view->getViewer();
+
+    // get all objects in the document
+    doc = Gui::Application::Instance->activeDocument()->getDocument();

```

Comment:
@donovaly This check is unsafe. Because if there is not active document then `activeDocument()` already returns null and thus will cause a segmentation fault.

Reply:
done in #6360